examples/tv-casting-app/tv-casting-common/src/CastingServer.cpp

```diff
@@ -44,9 +44,12 @@ void CastingServer::InitServer(std::function<CHIP_ERROR()> commissioningComplete
     mCommissioningCompleteCallback = commissioningCompleteCallback;
 
     // Enter commissioning mode, open commissioning window
-    static chip::CommonCaseDeviceServerInitParams initParams;
-    (void) initParams.InitializeStaticResourcesBeforeServerInit();
-    chip::Server::GetInstance().Init(initParams);
+    if (initChipServer)
```

Comment:
This function already has a `run only once` check above by using `mInited`. Some explanation of call paths may be helpful.

The comment on the if says "enter commissioning mode, open commissioning window". If this is true, the parameter should probably reflect that, like "openCommissioningWindow".

For readability sake `CastingServer::InitServer(callback, true)` and `CastingServer::InitServer(callback, false)` are not obviousl. I would recommend some enum/constant (and the constant should NOT be bool) so that code reads like `CastingServer::InitServer(callback, kOpenCommissioningWindow)`.

Reply:
I've moved these lines out to the main of the linux version of the app, as ChipAppServer.startApp() already takes care of doing this from android.